tests/integration/app-charm/pyproject.toml

```diff
@@ -0,0 +1,16 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+[tool.poetry]
+package-mode = false
+
+[tool.poetry.dependencies]
+python = "^3.10"
+ops = "^2.16.1"
+pymongo = "^4.8.0"
+kafka-python = "^2.0.2"
+kazoo = "^2.10.0"
+psycopg2 = "^2.9.9"
+mysql-connector-python = "9.0.0"
```

Comment:
`mysql-connector-python` must have the exact same version in poetry and `charm-binary-python-packages`, otherwise charmcraft's pip will try to update and rebuild it when installing to poetry source dependencies. IMHO it's easier to track an exact pin in both places.

Reply:
I didn't enable strict deps because the [ccraft 2 workaround](https://github.com/canonical/data-platform-workflows/blob/main/.github/workflows/build_charm.yaml#L146-L147) seems to be touching only in the base directory.

I'll try to tweak dpw and see if it helps, when I get a chance.

Reply:
Investigated further and the issue is that the charmcraft check expects the binary deps to be in the requirements, so just touching the file is not enough in this case. Sidestepped it by using charmcraft 3 for the test charm. IMHO it's not worth trying to fix this in dpw, since we'd be jumping to charmcraft 3 soon.